alb/terraform-examples/cluster-deployment/cluster-deployment.tf

```diff
@@ -0,0 +1,175 @@
+terraform {
+  required_providers {
+    vsphere = {
+      source  = "hashicorp/vsphere"
+      version = "2.0.2"
+     }    
+    avi = {
+      source  = "vmware/avi"
+      version = "21.1.2"
+    }
+  }
+}
+
+provider "vsphere" {
+  user                   = var.vsphere_user
+  password               = var.vsphere_password
+  vsphere_server         = var.vsphere_server
+  allow_unverified_ssl   = true
+}
+
+
+data "vsphere_resource_pool" "resource_pool" {
+  name                  = var.vsphere_resource_pool
+  datacenter_id 	= "${data.vsphere_datacenter.datacenter.id}"
+}
+
+data "vsphere_content_library" "content_library" {
+  name            = "AVI"
+}
+
+data "vsphere_content_library_item" "controllerovf" {
+  name        = "Controller"
+  type	      = "ovf"
+  library_id  = data.vsphere_content_library.content_library.id
+}
+
+resource "vsphere_virtual_machine" "controller" {
+  name              = var.Controller_vm_name[count.index]
+  resource_pool_id  = data.vsphere_resource_pool.resource_pool.id
+  datastore_id      =  "${data.vsphere_datastore.datastore.id}"
+  count             = 3
+  num_cpus          = 8
+  memory            = 24576
+  folder            = var.vm_folder
+  scsi_type         = "lsilogic"
+
+  network_interface {
+    network_id   = "${data.vsphere_network.management_net.id}"
+  }
+  lifecycle {
```

Comment:
May wish to ignore the `template_uuid`.

Reply:
The template UUID is required to specify the CL template.

Reply:
Yes, however, if the `termplate_uuid` ever changes then it will `ForceNew`.

```hcl
resource "vsphere_virtual_machine" "vm" {
  #... other configuration...
  }
  lifecycle {
    ignore_changes = [
      clone[0].template_uuid,
    ]
  }
}
```

Reply:
Im sorry I dont understand. We are retrieving the template when this is run. If the uuid changes, it will get the new uuid. This script would technically only be run once to deploy the templates.

Reply:
I'll let this one go, but it's a recommended practice.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.